File: CPP/Baju.cpp
#include <iostream>
#include <list>
#include <string>
#include "Aksesoris.cpp"

using namespace std;

class Baju : public Aksesoris {
    private:
        string jenis_hewan;
        string size;
        string merk;

    public:
        Baju() {
            jenis_hewan = "";
            size = "";
            merk = "";
        }

        Baju(int id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna, string jenis_hewan, string size, string merk) : Aksesoris(id, nama_produk, harga_produk, stok_produk, jenis, bahan, warna) { 
        this->jenis_hewan = jenis_hewan;
        this->size = size;
        this->merk = merk;
    }
    

        void setJenisHewan(string jenis_hewan) {
            this->jenis_hewan = jenis_hewan;
        }

        string getJenisHewan() {
            return jenis_hewan;
        }

        void setSize(string size) {
            this->size = size;
        }

        string getSize() {
            return size;
        }

        void setmerk(string merk) {
            this->merk = merk;
        }

        string getmerk() {
            return merk;
        }

        int jenisHewanLength() {
            return jenis_hewan.length();
        }

        int sizeLength() {
            return size.length();
        }

        int merkLength() {
            return merk.length();
        }
};

File: Baju.java
public class Baju extends Aksesoris {
    private String jenisHewan;
    private String size;
    private String merk;

    public Baju() {
        super();
        this.jenisHewan = "";
        this.size = "";
        this.merk = "";
    }

    public Baju(int id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna, String jenisHewan, String size, String merk) {
        super(id, namaProduk, hargaProduk, stokProduk, jenis, bahan, warna);
        this.jenisHewan = jenisHewan;
        this.size = size;
        this.merk = merk;
    }

    public String getJenisHewan() {
        return jenisHewan;
    }

    public void setJenisHewan(String jenisHewan) {
        this.jenisHewan = jenisHewan;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMerk() {
        return merk;
    }

    public void setMerk(String merk) {
        this.merk = merk;
    }
}
